writer.go

```diff
@@ -443,7 +443,7 @@ func (p *MediaPlaylist) Encode() *bytes.Buffer {
 
 	head := p.head
 	count := p.count
-	for i := uint(0); i < p.winsize && count > 0; count-- {
+	for i := uint(0); (i < p.winsize || p.winsize == 0) && count > 0; count-- {
```

Comment:
Sorry, you're right, it should have be written as:

```
    for i := uint(0); (p.Closed || i < p.winsize && count > 0); count-- {
```

The original proposal `(i < p.winsize || p.winsize == 0) && count > 0` would ignore the `winsize` when it is zero, but didn't express the fact we're doing this because VoD has a `winsize` of zero. So I proposed we switch this to `(i < p.winsize || p.Closed) && count > 0` which expresses the previous statement a little more clearly. But this would then ignore the `winsize` of *any* VoD content, which almost certainly isn't what anyone wants (it's OK for the moment as the `winsize` cannot be changed, but this behaviour may change in future versions).

So we need to check why the `winsize` is getting to 0 for VoD content in the first place, and perhaps have it set to `count` instead, as I believe that's where the root cause is (but I haven't looked closely at this either, seeing an example would help).